tests/common_setup.py

```diff
@@ -23,7 +23,11 @@
 
 class NeMoUnitTest(unittest.TestCase):
     def setUp(self) -> None:
-        nemo.core.neural_factory.NeuralModuleFactory.reset_default_factory()
-        print("---------------------------------------------------------")
-        print(self._testMethodName)
-        print("---------------------------------------------------------")
+        """ Default setup - instantiates Neural Factory. """
+        # Initialize the default Neural Factory - on CPU.
+        self.nf = nemo.core.NeuralModuleFactory(placement=nemo.core.DeviceType.CPU)
```

Comment:
Please do not set tests to CPU. Naturally, we care much, much more about GPUs :)

Reply:
so all tests should run on GPU.
We can add more tests running on CPU later, but this is lower priority and outside of this PR

Reply:
Generally I agree that we should care about GPUs.

But what are we testing here? GPUs? CUDA? Or software that can run on both CPUs and GPU? If it passes on CPU, then the rest remain to the underlying libraries, not really NeMo.

Besides, my problem with running unittests on GPU is that when I am working on Mac I cannot run them at all - and I am not the only person that uses Macs, right? :]

I will revert that, but I leave the subject open, think we should discuss that along with how and what are we really testing, there are many "unit tests" that are really not unit, moreover, plenty of them are simply incorrect and not working...